### DISCUSSION TITLE
Installation - Memory Error:Fatal error: Allowed memory size of 1610612736 bytes exhausted

### DISCUSSION THREAD

**Discussion by nobicycle** [spatie/laravel-backup#1093] [Q&A] [answered]
Hello and thank you for laravel-backup that I am installing for the first time.

Initially php memory_limit = 128M

```
$ php -v
PHP 7.4.6 (cli) (built: May 12 2020 15:48:23) ( NTS )

```

```
$ composer -v require spatie/laravel-backup
Using version ^6.11 for spatie/laravel-backup
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
PHP Fatal error:  Allowed memory size of 1610612736 bytes exhausted (tried to allocate 4096 bytes) in phar:///usr/bin/composer/src/Composer/DependencyResolver/Solver.php on line 223

Fatal error: Allowed memory size of 1610612736 bytes exhausted (tried to allocate 4096 bytes) in phar:///usr/bin/composer/src/Composer/DependencyResolver/Solver.php on line 223

Check https://getcomposer.org/doc/articles/troubleshooting.md#memory-limit-errors for more info on how to handle out of 

I then increased memory to 256M 
```

$ php -r "echo ini_get('memory_limit').PHP_EOL;"
256M

$ composer -v require spatie/laravel-backup
Using version ^6.11 for spatie/laravel-backup
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
PHP Fatal error:  Allowed memory size of 1610612736 bytes exhausted (tried to allocate 4096 bytes) in phar:///usr/bin/composer/src/Composer/DependencyResolver/Solver.php on line 223

Check https://getcomposer.org/doc/articles/troubleshooting.md#memory-limit-errors for more info on how to handle out of memory errors.

```

**Comment by RishabSwift** [comment]
Did you figure it out? Having the same issue.

**Reply by daud123333** [reply]
same problem

**Comment by kumarkishan070997** [comment]
i am also having the same issue

**Reply by wisegear** [reply]
The solution is to increase the amount of memory, Composer can be quite intensive.  I have found that you need to allocate 1GB of ram to be sure that you do not get an out of memory error.  If you are using a VPS that does not have that much then try increasing or adding SWAP.

**Comment by nobicycle** [comment]
Hi,

From my   message above:

_Fatal error: Allowed memory size of **1610612736** bytes exhausted (tried to allocate **4096** bytes)_ 

I did the test again
```
php72 -r "echo ini_get('memory_limit').PHP_EOL;"
1024M

```
And again ...
```

$ sudo -u http /bin/bash -c "cd ~/delivery; composer require spatie/laravel-backup"
Using version ^6.11 for spatie/laravel-backup
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
PHP Fatal error:  Allowed memory size of **1610612736** bytes exhausted (tried to allocate **134217736** bytes) in phar:///usr/bin/composer/src/Composer/DependencyResolver/RuleSet.php on line 84

```
0.134217736 bytes is 0.13 GB

whereas 

1610612736 bytes is 1.6 GB

Allowed php memory is much greater.
Does anything else need changing in php.ini apart from memory_limit?

**Reply by wisegear** [reply]
Having 1.6GB allocated and it being available for use are not the same.  How much memory does your system/server actually have?  It is telling you the memory is being exhausted, so you need to look at what memory is in use when you run it, what is left and whether that is enough.  Clearly it isn't but you need to look more closely at the resource usage.

**Comment by nobicycle** [comment]
With 2.2GB free I still get the problem.

```
$ free -h
              total        used        **free**      shared  buff/cache   available
Mem:          7.5Gi       2.3Gi       **2.2Gi**       384Mi       2.9Gi       4.4Gi
Swap:            0B          0B          0B

[user@tower ~]$ php72 -r "echo ini_get('memory_limit').PHP_EOL;"
2024M
[user@tower ~]$ free -h
              total        used        free      shared  buff/cache   available
Mem:          7.5Gi       2.4Gi       2.2Gi       401Mi       2.9Gi       4.4Gi
Swap:            0B          0B          0B
[user@tower ~]$ sudo -u http /bin/bash -c "cd ~/delivery;php72 /usr/bin/composer require spatie/laravel-backup"
Using version ^6.11 for spatie/laravel-backup
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
PHP Fatal error:  Allowed memory size of 2122317824 bytes exhausted (tried to allocate 4096 bytes) in phar:///usr/bin/composer/src/Composer/DependencyResolver/Solver.php on line 223

Fatal error: Allowed memory size of 2122317824 bytes exhausted (tried to allocate 4096 bytes) in phar:///usr/bin/composer/src/Composer/DependencyResolver/Solver.php on line 223

```

**Comment by redboxsk** [comment, marked as answer]
Hello, try:
COMPOSER_MEMORY_LIMIT=-1 composer -v require spatie/laravel-backup

I have used it during installation of another package with success

**Reply by nobicycle** [reply]
Thank you. That works!

**Reply by rachmadbud** [reply]
I got an error
`COMPOSER_MEMORY_LIMIT : The term 'COMPOSER_MEMORY_LIMIT' is not recognized as the name of a cmdlet, function, 
script file, or operable program. Check the spelling of the name, or if a path was included, verify that the 
path is correct and try again.
At line:1 char:1
+ COMPOSER_MEMORY_LIMIT = -1 komposer -v membutuhkan spatie / laravel-b ...
+ ~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (COMPOSER_MEMORY_LIMIT:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException`

**Reply by fulix99** [reply]
También me saltó el mismo error 
COMPOSER_MEMORY_LIMIT : El término 'COMPOSER_MEMORY_LIMIT' no se reconoce como nombre de un cmdlet, función, archivo de script o programa ejecutable. Compruebe 
si escribió correctamente el nombre o, si incluyó una ruta de acceso, compruebe que dicha ruta es correcta e inténtelo de nuevo.
En línea: 1 Carácter: 1
+ COMPOSER_MEMORY_LIMIT = -1 composer -v require Spatie / laravel-backu ...
+ ~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (COMPOSER_MEMORY_LIMIT:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

**Reply by fulix99** [reply]
Sin embargo ya pude solucionarlo!

Pueden hacerlo manualmente yendo a xampp -> php -> php.ini -> buscar: memory_limit -> cambiar el valor a -1 (en mi caso era 512M) y luego cierran y abren el xampp. Habiendo hecho eso ya pueden instalar el paquete que querían instalar (en mi caso era nesbot/carbon)

**Comment by Abdullahmasood553** [comment]
Open your xampp.
Go to actions and press config.
Open php.ini file
Change
memory_limit=512M
to  
memory_limit= -1

**Reply by abdelrahmanali34** [reply]
thanks very match

**Reply by mohrezaaaa** [reply]
thanks !
